specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1134 @@
+
+## 4.6. PackageVolume.RepairPackageVolume()
+
+Fabrikam app checks if a PackageVolume is OK and if not, prompts the user to confirm it should
+proceed to repair the PackageVolume.
+
+```c#
+var CheckAndFixPackageVolume(string path)
+{
+    var packageVolume = PackageVolumeManager.FindPackageVolumeByPath(path);
+    if (!packageVolume.CheckPackageVolume())
+    {
+        bool ok = PromptUserForConfirmation();
+        if (ok)
+        {
+            try
+            {
+                packageVolume.RepairPackageVolume();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+            }
+        }
+    }
+}
+```
+
+# 5. Remarks
+
+## 5.1. Platform Support
+
+This API is only available on Windows >= 10.0.19041.0 (aka 2004 aka 20H1).
+
+A subset of functionality is available on newer releases, e.g. `AddPackageOptions.ExpectedDigests`
+requires Windows >= 10.0.22621.0 (aka Win11 22H2). Any
+functionality requiring newer releases of windows than then 20H1 baseline has affordances to detect at runtime if the current platform supports the feature e.g.
+
+```c#
+var options = new AddPackageOptions();
+if (options.IsLimitToExistingPackagesSupported)
+{
+    options.LimitToExistingPackages = true;
+}
+```
+
+# 6. API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Windows.Management.Deployment
+{
+    [contractversion(1)]
+    apicontract PackageDeploymentContract{};
+
+    /// Represents a package storage volume.
+    /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageVolume
+    {
+        PackageVolume();
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.issystemvolume
+        Boolean IsSystemVolume{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.mountpoint
+        String MountPoint{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.name
+        String Name{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.packagestorepath
+        String PackageStorePath{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.supportshardlinks
+        Boolean SupportsHardLinks{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.isfulltrustpackagesupported
+        Boolean IsFullTrustPackageSupported{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.isappxinstallsupported
+        Boolean IsAppxInstallSupported{ get; };
+
+        /// Return true if the package volume is OK, else false if it needs to be repaired.
+        bool CheckPackageVolume();
```

Comment:
Since this is a method (not property) I presume it's doing work when called to perform the 'check'.  There is precedence for GetIs* method naming.  Would we prefer e.g. `GetIsPackageVolumeHealthy` ?